src/waveform/renderers/waveformrendermark.cpp

```diff
@@ -50,7 +50,7 @@ void WaveformRenderMark::draw(QPainter* painter, QPaintEvent* /*event*/) {
             generateMarkImage(mark.data());
         }
 
-        int samplePosition = mark->m_pPointCos->get();
+        int samplePosition = mark->getPlayPosition();
```

Comment:
Ups ... The playpoition is a samplePosition, so the range is not 0.0 ... 1.0 it is up to samplePosition. 
So we need to rename the function to getSamplePoition() or just samplePosition()

We also have here a rounding issue double to int.
This is an other task to fix this and allow to set marks between two samples.  
This can happen if you play a track at a very low speed.